### PR TITLE
[INFINITY-1649] Add SDK CLI unit tests to build

### PR DESCRIPTION
**Pull request by paegun**
* Add SDK CLI unit tests to build.

NOTE: use of pushd/popd is the first use w/i this script, but they are used elsewhere w/i the build process, i.e. via shakedown scp.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -69,10 +69,20 @@ if [ x$PULLREQUEST = "xtrue" ]; then
   fi
 fi
 
-# Build steps for SDK libraries:
+# Verify SDK CLI, run unit tests
+_notify_github pending "SDK CLI build running"
+pushd $REPO_ROOT_DIR/cli
+go test ./...
+if [ $? -ne 0 ]; then
+  _notify_github failure "SDK CLI build failed"
```
Comment (jsrodman) [inline build.sh]: This message seems misleading (build worked, test failed)? But it doesn't matter since we're not relying on _notify_github anymore.

**Comment by NimaVaziri** [comment]
@paegun is this branch rebased on master? the fix for HDFS failures occurring has been merged

File: build.sh (hunk 1 above)
Comment (elezar) [inline build.sh]: Quick question @jsrodman. At which point are the dependencies for this installed? Testing this locally (in the docker container), I get:
```
./build.sh
[STATUS] build:sdk pending: SDK CLI build running
/build/cli /build
commands.go:12:2: cannot find package "github.com/mesosphere/dcos-commons/cli/client" in any of:
	/usr/lib/go-1.8/src/github.com/mesosphere/dcos-commons/cli/client (from $GOROOT)
	/go-tmp/src/github.com/mesosphere/dcos-commons/cli/client (from $GOPATH)
commands.go:13:2: cannot find package "github.com/mesosphere/dcos-commons/cli/commands" in any of:
	/usr/lib/go-1.8/src/github.com/mesosphere/dcos-commons/cli/commands (from $GOROOT)
	/go-tmp/src/github.com/mesosphere/dcos-commons/cli/commands (from $GOPATH)
commands.go:14:2: cannot find package "github.com/mesosphere/dcos-commons/cli/config" in any of:
	/usr/lib/go-1.8/src/github.com/mesosphere/dcos-commons/cli/config (from $GOROOT)
	/go-tmp/src/github.com/mesosphere/dcos-commons/cli/config (from $GOPATH)
commands.go:15:2: cannot find package "gopkg.in/alecthomas/kingpin.v3-unstable" in any of:
	/usr/lib/go-1.8/src/gopkg.in/alecthomas/kingpin.v3-unstable (from $GOROOT)
	/go-tmp/src/gopkg.in/alecthomas/kingpin.v3-unstable (from $GOPATH)
[STATUS] build:sdk failure: SDK CLI build failed
/build
```

**Comment by jsrodman** [comment]
I assumed it operated. Since this must work for someone, it's probably a
dependency on the symlinks out of the goroot back to the source tree. These
are created in the build cli shell script.

Nick sun encountered a scenario where some type of link existed to fool the
script but not the intended one.

I'll look at it a bit when I get to a laptop usable location.

**Comment by elezar** [comment]
Yes, this was a fresh clone of the repo, so it's pretty likely that nothing was there. Thought I'd help to get your PRs across the line ;)

**Comment by paegun** [comment]
@elezar @jsrodman the golang 1.8 build dependency is listed in https://mesosphere.github.io/dcos-commons/developer-guide.html . The specific steps to install are elided. I use Go Version Manager (https://github.com/moovweb/gvm , gvm) for the reasons listed there. I did consider not executing the CLI unit tests if `which go` errors, but in that case the development environment cannot produce the CLI, so is invalid.

**Comment by elezar** [comment]
@paegun Seems like this is finally green! :shipit: (why did the state just change!?) 

Thanks for your feedback regarding the go dependency. For the record, I think it was probably me trying to do things that were not the original intent of this PR. I followed the following steps (in order to check this behaviour):

1. Edit `test.sh` to drop me into an interactive shell (i.e. add a `-ti` option and change the command to just `bash` where we execute the `docker run` command)
2. Run `./test.sh DUMMY` to start the container
3. Run `./build.sh` within the container

This then resulted in the error messages that I mentioned above.

I suppose my confusion comes from not knowing where `./build.sh` is used as part of our CI system (I only see it invoked from `./release.sh` and was thus wondering where it is exercised).

**Pull request merged**